src/blocks.cpp

```diff
@@ -2396,44 +2396,71 @@ void PSwitch(bool enabled)
 
 void PowBlock()
 {
+    // For NetPlay: make this only affect screens where the earthquake block NPC is near at least one vScreen
+
     int numScreens = 0;
-    int A = 0;
     int Z = 0;
 
     PlaySound(SFX_Stone);
-    numScreens = 1;
 
-    if(!LevelEditor)
+    // Shake blocks. This isn't just cosmetic, it damages NPCs, so we should check the canonical screens here.
+
+    for(int screen_i = 0; screen_i < c_screenCount; screen_i++)
     {
-        if(ScreenType == 1)
-            numScreens = 2;
-        if(ScreenType == 4)
-            numScreens = 2;
+        Screen_t& screen = Screens[screen_i];
 
-        if(ScreenType == 5)
+        // use modern screen iteration bounds
+        if(g_compatibility.allow_multires)
         {
-            DynamicScreen(Screens[0]);
-            if(vScreen[2].Visible)
-                numScreens = 2;
-            else
-                numScreens = 1;
+            Z = screen.active_begin() + 1;
+            numScreens = screen.active_end();
         }
-    }
+        // buggy original code
+        else
+        {
+            Z = 1;
+            numScreens = 1;
 
-    for(Z = 1; Z <= numScreens; Z++)
-    {
-        for(A = 1; A <= numBlock; A++)
+            if(!LevelEditor)
+            {
+                if(ScreenType == 1)
+                    numScreens = 2;
+                if(ScreenType == 4)
+                    numScreens = 2;
+
+                if(ScreenType == 5)
+                {
+                    DynamicScreen(screen);
+                    if(vScreen[2].Visible)
```

Comment:
Maybe use a ternary expression here?
```cpp
numScreens = vScreen[2].Visible ? 2 : 1;
```

Reply:
Good note, I wanted to keep the exact format since this is copied legacy logic. Do you still want me to update it?

Reply:
Ye, it would be better to update

Reply:
I updated the syle of this section (removed the `else` clause and combined the previous `if` statements)